department-of-veterans-affairs/abd-vro
# Create feature toggle service

## What was the problem?
adds rabbitmq callback + redis hash for feature toggle service. There's no endpoint set up on the java/camel side yet, so no testing there yet. It will be added as part of #802 

Associated tickets or Slack threads:

- [VRO-245](https://github.com/department-of-veterans-affairs/abd-vro/issues/245)

## How does this fix it?[^1]

- adds featuretoggle image to image-names.sh
- adds code for parsing yml file with feature flags
- adds parsed data to redis hash / rabbitmq callback

## How to test this PR
- pull branch (`tanner/feature-toggle-service`) and build your containers. a featuretoggle container should be built without issue
- visit http://localhost:15672/#/queues and see that the queue/exchange for this service exists

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
